homeassistant/components/media_player/sonos.py

```diff
@@ -714,12 +733,15 @@ def source_list(self):
     @property
     def source(self):
         """Name of the current input source."""
-        if self._is_playing_line_in:
-            return SUPPORT_SOURCE_LINEIN
-        if self._is_playing_tv:
-            return SUPPORT_SOURCE_TV
-
-        return None
+        if self._coordinator:
+            return self._coordinator.source
+        else:
+            if self._is_playing_line_in:
+                return SUPPORT_SOURCE_LINEIN
+            elif self._is_playing_tv:
+                return SUPPORT_SOURCE_TV
+            else:
+                return None
```

Comment:
Super minor style nit, but the final `else` block isn't even necessary. I think it looks a bit nicer to just left indent it twice and drop the else since you can't continue execution after any return statements. I call it a fall through return and tends to be best for long term maintenance when other people touch the code.